pkg/platform/api/headchef/headchef.go

```diff
@@ -132,10 +131,16 @@ func (b *BuildParams) WriteToRequest(req runtime.ClientRequest, reg strfmt.Regis
 }
 
 func (r *Client) reqBuild(buildReq *headchef_models.V1BuildRequest, buildStatus *BuildStatus) {
+	defClient := retryhttp.DefaultClient
+	timeout := defClient.MaxTimeout(retryhttp.DefaultTimeout)
+
+	ctx, cancel := retryhttp.NewContext(nil, timeout)
+	defer cancel()
+
 	startParams := headchef_operations.StartBuildV1Params{
-		Context:      context.Background(),
+		Context:      ctx,
 		BuildRequest: buildReq,
-		HTTPClient:   retryhttp.DefaultClient.StandardClient(),
+		HTTPClient:   defClient.StandardClient(),
```

Comment:
This still looks more complicated than it needs to be in my eyes.

Can we simplify to something like:

```
	client := retryhttp.NewClient(60, 3)
	defer client.Close()

	startParams := headchef_operations.StartBuildV1Params{
		Context:      client.Context,
		BuildRequest: buildReq,
		HTTPClient:   client.HTTPClient,
	}
```

We want retryhttp to own as much of this as possible.

Reply:
The context.Context type is meant to be a per-request payload. For instance, in the net/http package, the Context type is not included as a field in the Client type, but is injected via a Request type field when calling the Client's Do method as opposed to the Get/Post/Head methods. I think it's very likely that those convenience methods would include a context.Context param if the context package had been part of the stdlib at that time net/http was authored.

My concern here is that we are adding to the complexity/conflation that is building as with the timeout wrangling.

Reply:
After speaking in Slack, we're going with creating a containing type called RetryHTTP to avoid any confusion about Request-specific data.